pkg/controllers/vdb/onlineupgrade_reconciler.go

```diff
@@ -426,12 +424,17 @@ func (o *OnlineUpgradeReconciler) processSecondary(ctx context.Context, sts *app
 
 // isMatchingSubclusterType will return true if the subcluster type matches the
 // input string.  Always returns false for the transient subcluster.
-func (o *OnlineUpgradeReconciler) isMatchingSubclusterType(sts *appsv1.StatefulSet, scType string) (bool, error) {
-	isTransient, err := strconv.ParseBool(sts.Labels[vmeta.SubclusterTransientLabel])
-	if err != nil {
-		return false, fmt.Errorf("could not parse label %s: %w", vmeta.SubclusterTransientLabel, err)
+func (o *OnlineUpgradeReconciler) isMatchingSubclusterType(sts *appsv1.StatefulSet, scType string) bool {
+	stsScType := sts.Labels[vmeta.SubclusterTypeLabel]
+	if stsScType != scType {
+		return false
+	}
+
+	transientName, hasTransient := o.Vdb.GetTransientSubclusterName()
+	if !hasTransient {
+		return true
 	}
-	return sts.Labels[vmeta.SubclusterTypeLabel] == scType && !isTransient, nil
+	return sts.Labels[vmeta.SubclusterNameLabel] != transientName
```

Comment:
I know it is been a while but why dowe need to return false for transient sc?

Reply:
When online upgrade iterates over subclusters (i.e. to upgrade, add client routing, etc), we always want to skip transient. Those aren't ever upgraded. They are meant to only serve read requests during the upgrade.